File: Java/0001-0050/11. Container With Most Water.java
// https://leetcode.com/problems/container-with-most-water/

class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length-1;
        int maxArea = 0;

        while(left < right){
            int currentArea = Math.min(height[left], height[right]) * (right - left);
            maxArea = Math.max(maxArea, currentArea); // update maxArea

            //Update Pointers
            if(height[left] < height[right]) left++;
            else right--;
        }

        return maxArea;
    }
}

File: Python/0001-0050/11. Container With Most Water.py
# https://leetcode.com/problems/container-with-most-water/

class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height)-1
        maxArea = 0

        while left < right:
            currentArea = min(height[left], height[right]) * (right - left)
            maxArea = max(maxArea, currentArea) # update maxArea

            # Update Pointers
            if height[left] < height[right]:
                left += 1
            else: 
                right -= 1

        return maxArea
